json struct.py

```diff
@@ -101,11 +101,12 @@ def recursive_generate_cpp_content(root, configuration, chain = ''):
     plug_text = ''
     for key, value in configuration:
         if isinstance(value, dict) or isinstance(value, list):
+            old_chain = chain
```

Comment:
Why do we need this elaborate? Each iteration has its own parameters and we're not modifying chain, just new_chain.
I looked further and you seem to think this solves subsections better by not adding .general where it doesn't belong. 
I looked at my implementation of vegastrike_config.json and it doesn't seem to be adding general where it shouldn't. Can you explain this further and maybe show an example where the old code is failing?

Reply:
I tried running your code using the sample config JSON provided, but with:

```
(text not indented)
    (text indented one level)
        (text indented two levels)
            (text indented three levels)
        (text indented two levels) (this bit added)
```

At the final line, indented a bit less, your code fell over. It didn't unindent the last section properly. That's why I made those changes to `json struct.py`.